src/SimpleCLIUserInterface.cpp

```diff
@@ -0,0 +1,352 @@
+// Copyright (C) Sven Rosvall (sven@rosvall.ie)
+// This file is part of VLCB-Arduino project on https://github.com/SvenRosvall/VLCB-Arduino
+// Licensed under the Creative Commons Attribution-NonCommercial-ShareAlike 4.0 International License.
+// The full licence can be found at: http://creativecommons.org/licenses/by-nc-sa/4.0/
+
+#include "SimpleCLIUserInterface.h"
+#include "Controller.h"
+#include <Streaming.h>
+#include <SimpleCLI.h>
+
+extern void printConfig();
+
+
+namespace VLCB
+{
+
+// Create CLI Object
+SimpleCLI simpleCli(CLI_COMMAND_QUEUE_SIZE); // Default of 10 commands
+
+Command cmdHelp;
+Command cmdHelpWithArg;
+Command cmdSerialInterface;
+
+//////////////////////////////////////////////////////////////////
+// See Modern C++ Design, Andrei Alexandrescu, 2001(!) page 116
+// First declare a typedef for a pointer to what is needed.
+// Note the argument type.
+typedef void (SimpleCLIUserInterface::* TpMemFun)(cmd *);
+// Then create an object which implements tha callback
+TpMemFun pSerialCallback = &SimpleCLIUserInterface::serialCallback;
```

Comment:
Please make pSerialCallback "static" to reduce scope to this file.

Reply:
I think that the static version does not work - see comments in the example and previous email conversations.